File: spikes/02-dynamic-loading/plugins/example_plugin.zig
const std = @import("std");

/// Plugin information - must match the API version
const plugin_info = PluginInfo{
    .api_version = 1,
    .name = "example_plugin",
    .display_name = "Example Plugin",
    .version = "1.0.0",
};

const PluginInfo = extern struct {
    api_version: u32,
    name: [*:0]const u8,
    display_name: [*:0]const u8,
    version: [*:0]const u8,
};

export fn plugin_get_info() *const PluginInfo {
    return &plugin_info;
}

export fn plugin_init() void {
    std.debug.print("[ExamplePlugin] Initialized!\n", .{});
}

export fn plugin_deinit() void {
    std.debug.print("[ExamplePlugin] Deinitialized!\n", .{});
}

export fn plugin_render_menu() void {
    std.debug.print("[ExamplePlugin] Rendering menu\n", .{});
}

export fn plugin_render_panel() void {
    std.debug.print("[ExamplePlugin] Rendering panel\n", .{});
}

File: spikes/03-lua-scripting/scripts/example_plugin.lua
-- Example Lua Plugin for Labelle
-- This demonstrates how a Lua script can define panels and menus

Plugin = {
    name = "lua_example",
    display_name = "Lua Example Plugin",
    version = "1.0.0"
}

-- Called when the plugin is loaded
function Plugin.on_init()
    print("[LuaPlugin] Initialized!")
end

-- Called when the plugin is unloaded
function Plugin.on_deinit()
    print("[LuaPlugin] Deinitialized!")
end

-- Called to render menu items
function Plugin.render_menu()
    print("[LuaPlugin] Rendering menu")
    -- In real implementation, this would call imgui functions
    -- imgui.menu_item("My Lua Tool")
end

-- Called to render panel content
function Plugin.render_panel()
    print("[LuaPlugin] Rendering panel")
    -- In real implementation, this would call imgui functions
    -- imgui.text("Hello from Lua!")
    -- if imgui.button("Click me") then
    --     print("Button clicked!")
    -- end
end

return Plugin
